scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,409 @@
+
+    //Default value is getLocalSpace() because it will always be writable and moreover can be used to transfer files to another data space (global, user)
+    private String ftpLocalFilePath = getLocalSpace();
+
+    private String ftpRemoteFilePath = null;
+
+    private String ftpMode = null;
+
+    private String ftpUsername = null;
+
+    private String ftpPassword = null;
+
+    private static final String FTP_LOCAL_FILE_PATH = "ftpLocalFilePath";
+
+    /**
+     * @see JavaExecutable#init(Map)
+     */
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("ftpHostname")) {
+            ftpHostname = args.get("ftpHostname").toString();
+        }
+        if (args.containsKey("ftpPort")) {
+            ftpPort = Integer.parseInt(args.get("ftpPort").toString());
+        }
+        if (args.containsKey("ftpExtractArchive")) {
+            ftpExtractArchive = Boolean.parseBoolean(args.get("ftpExtractArchive").toString());
+        }
+        if (args.containsKey(FTP_LOCAL_FILE_PATH) && args.get(FTP_LOCAL_FILE_PATH).toString() != null) {
+            ftpLocalFilePath = args.get(FTP_LOCAL_FILE_PATH).toString();
+        }
+        if (args.containsKey("ftpRemoteFilePath")) {
+            ftpRemoteFilePath = args.get("ftpRemoteFilePath").toString();
+        }
+        if (args.containsKey("ftpMode")) {
+            ftpMode = args.get("ftpMode").toString();
+        }
+        ftpUsername = getThirdPartyCredential("FTP_USERNAME");
+        ftpPassword = getThirdPartyCredential("FTP_PASSWORD");
+        if (ftpUsername == null || ftpPassword == null) {
+            throw new IllegalArgumentException("You first need to add your ftp username and password (FTP_USERNAME, FTP_PASSWORD) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException {
+        List<String> filesRelativePathName;
+        FTPClient ftpClient = new FTPClient();
+        // connect and login to the server
+        ftpClient.connect(ftpHostname, ftpPort);
+        int reply = ftpClient.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftpClient.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+        ftpClient.login(ftpUsername, ftpPassword);
+
+        // use local passive mode to pass firewall
+        ftpClient.enterLocalPassiveMode();
+
+        getOut().println("Connected");
+
+        switch (ftpMode) {
+            //FTP mode is GET
+            case "GET":
+                filesRelativePathName = FtpGet(ftpClient);
+                break;
+
+            //FTP mode is PUT
+            case "PUT":
+                filesRelativePathName = FtpPut(ftpClient);
+                break;
+
+            default:
+                throw new RuntimeException("wrong value");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

Reply:
sonar is right. IllegalArgumentException, with an understandable error message (which parameter is wrong)

Reply:
okey